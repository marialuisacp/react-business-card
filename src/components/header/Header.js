import React, { Component } from 'react';
import '../../assets/styles/styles.scss';
import './Header.scss';

class Header extends Component {
  render() {
    return (
      <div id='header'>
        <div className='content'>
          <img alt='Logo do projeto' src='assets/images/logo.png' className='logo' />
        </div>
        <div className="lines">
          <div className='column column-border column-left'></div>
          <div className='column column-center' style={{ backgroundImage: 'url(assets/images/lines.png)' }}></div>
          <div className='column column-border'></div>
        </div>
      </div>
    );
  }
}

export default Header;