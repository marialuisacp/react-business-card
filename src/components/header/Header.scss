#header {
  $column-center-width: 276px;
  $logo-width: 150px;
  
  position: relative;
  width: 100%;
  // z-index: 9;

  .logo {
    width: $logo-width;
    max-width: $logo-width;
    display: block;
  }

  .lines {
    display: flex;
    width: 100%;
    height: 150px;
    
    .column {
      display: inline-flex;
      height: 150px;
    }
    .column-border {
      width: calc((100% - 276px) / 2);
    }
    .column-left {
      border-top: solid 2px #fff;
    }
    .column-center {
      width: $column-center-width;
      background-image: url(/images/lines.png);
      background-repeat: no-repeat;
      background-size: 100%;
      background-position: center top;
    }
  }
}

@media (max-width: 1080px) {
  #header {
    .lines {
      .column-border {
        width: 20%;
      }
      .column-center {
        width: 60%;
      }
    }
  }
}