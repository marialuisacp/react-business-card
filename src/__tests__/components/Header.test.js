import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../components/Header/Header';

describe('Header', () => {
  it('Component Header should render correctly', () => {
    const component = renderer.create(
      <Header />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
