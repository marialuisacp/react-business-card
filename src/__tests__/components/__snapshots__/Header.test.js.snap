// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header should render correctly 1`] = `
<div
  id="header"
>
  <div
    className="content"
  >
    <img
      alt="Logo do projeto"
      className="logo"
      src="assets/images/logo.png"
    />
  </div>
  <div
    className="lines"
  >
    <div
      className="column column-border column-left"
    />
    <div
      className="column column-center"
      style={
        Object {
          "backgroundImage": "url(assets/images/lines.png)",
        }
      }
    />
    <div
      className="column column-border"
    />
  </div>
</div>
`;

exports[`Header Component Header should render correctly 1`] = `
<div
  id="header"
>
  <div
    className="content"
  >
    <img
      alt="Logo do projeto"
      className="logo"
      src="assets/images/logo.png"
    />
  </div>
  <div
    className="lines"
  >
    <div
      className="column column-border column-left"
    />
    <div
      className="column column-center"
      style={
        Object {
          "backgroundImage": "url(assets/images/lines.png)",
        }
      }
    />
    <div
      className="column column-border"
    />
  </div>
</div>
`;
